controllers/serviceinstance_controller.go

```diff
@@ -120,13 +124,69 @@ func (r *ServiceInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return r.createInstance(ctx, smClient, serviceInstance)
 	}
 
+	// Handle instance share change if needed
+	if instanceShareChangeNeedsToBeHandled(serviceInstance) {
+		if err := r.handleInstanceSharingChange(ctx, serviceInstance, smClient); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true, RequeueAfter: r.Config.PollInterval}, nil
```

Comment:
why requeue is required?

Reply:
because we may need to update 
for example someone shared and immediately updated

Reply:
it will never get to the update since you will return in this section:
	if isFinalState(serviceInstance) {
		log.Info(fmt.Sprintf("Final state, spec did not change, and we are not in progress - ignoring... Generation is - %v", serviceInstance.Generation))
		return ctrl.Result{}, nil
	}

Reply:
Why? the spec did changed because we did not update

Reply:
how can you tell what was changed in the spec when you are in reconcile?